src/graph_compute/phase_scheduling.h

```diff
@@ -117,7 +128,7 @@ static void createChunkData(vertex_t * const nodes, const vid_t cntNodes,
   cilk_for (vid_t i = 0; i < scheddata->cntChunks; ++i) {
     chunkdata_t * chunk = &scheddata->chunkdata[i];
     chunk->nextIndex = i << CHUNK_BITS;
-    chunk->phaseEndIndex[0] = std::min(chunk->nextIndex + (1 << (CHUNK_BITS - 1)),
+    chunk->phaseEndIndex[0] = std::min(chunk->nextIndex + ((1 << CHUNK_BITS) >> 1),
```

Comment:
What does setting CHUNK_BITS to zero mean? Shouldn't that be a nonsensical setting that should generate a compilation error? 



—
Sent from Mailbox

On Thu, Nov 5, 2015 at 1:12 PM, whasenpl <notifications@github.com> wrote:

>> @@ -117,7 +128,7 @@ static void createChunkData(vertex_t * const nodes, const vid_t cntNodes,
>>    cilk_for (vid_t i = 0; i < scheddata->cntChunks; ++i) {
>>      chunkdata_t * chunk = &scheddata->chunkdata[i];
>>      chunk->nextIndex = i << CHUNK_BITS;
>> -    chunk->phaseEndIndex[0] = std::min(chunk->nextIndex + (1 << (CHUNK_BITS - 1)),
>> +    chunk->phaseEndIndex[0] = std::min(chunk->nextIndex + ((1 << CHUNK_BITS) >> 1),
> CHUNK_BITS can be 0, which generates a compilation error.  The new version doesn't generate an error and is equivalent in meaning when CHUNK_BITS is greater than 0.
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/obi1kenobi/laika/pull/1/files#r44046623